File: StreamProviderWS/StreamProviderWS/Models/Common/ChatMessage.cs
using System;

namespace StreamProviderWS.Models.Common
{
    public class ChatMessage : Base
    {
        public string RoomId { get; set; }

        public User User { get; set; }

        public string Message { get; set; }

        public string VoiceMessage { get; set; }

        public DateTime DateTime { get; set; }

        public double MovieCurrentTime { get; set; } // seconds into the movie
    }
}

File: VideoStreamerUI/src/app/entities/chatMessage.ts
import { SocialUser } from 'angularx-social-login';

export class ChatMessage {
    Id: string;
    RoomId: string;
    User: SocialUser;

    Message: string;
    VoiceMessage: string;
    DateTime: Date;
    MovieCurrentTime: number; // seconds into the movie

    constructor(roomId: string, user: SocialUser, message: string, voiceMessage: string, datetime: Date, movieCurrentTime: number) {
        this.RoomId = roomId;
        this.User = user;
        this.Message = message;
        this.VoiceMessage = voiceMessage;
        this.DateTime = datetime;
        this.MovieCurrentTime = movieCurrentTime;
    }
}
